### PR TITLE
chore(Cross): [IOAPPX-210] Upgrade `react-navigation` to v6

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR upgrades react-navigation to `6.x` with all its peer dependencies

**refs:** 
- https://reactnavigation.org/docs/upgrading-from-5.x/
- https://reactnavigation.org/docs/troubleshooting/

## List of changes proposed in this pull request
- Version upgrade
- Components refactoring to handle the deprecated navigation prop

## How to test
```
rm -rf node_modules
cd ios && bundle exec pod deintegrate && cd .. 
yarn
cd ios && bundler exec pod install && cd .. 
yarn react-native start --reset-cache
yarn reat-native run-yourplatform
```
Run tests and check all of the main app flows

### CODE REVIEW

File: ts/features/walletV3/payment/navigation/navigator.tsx [hunk 1]
```diff
@@ -20,8 +20,7 @@ const Stack = createStackNavigator<WalletPaymentParamsList>();
 export const WalletPaymentNavigator = () => (
   <Stack.Navigator
     initialRouteName={WalletPaymentRoutes.WALLET_PAYMENT_MAIN}
-    headerMode={"screen"}
-    screenOptions={{ gestureEnabled: isGestureEnabled }}
+    screenOptions={{ gestureEnabled: isGestureEnabled, headerShown: true }}
```
Comment (CrisTofani) [inline ts/features/walletV3/payment/navigation/navigator.tsx:23]: @mastro993  check here
```suggestion
    screenOptions={{ gestureEnabled: isGestureEnabled, headerMode: true }}
```

**Review by LazyAfternoons — APPROVED** [review]
LGTM for FCI related flows.

**Review by mastro993 — APPROVED** [review]
Tested on Bonus, Wallet and Payment flows, everything works as expected!

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]
Problems with:
```
"react-native-pdf": "^6.4.0",
"react-native-pdf-thumbnail": "^1.2.1",
```
If we remove `yarn.lock`

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]
Bumping `react-native-screens` to `3.29.0` has RootModal(s) side effects... (iOS, still not tested on Android)

<video src="https://github.com/pagopa/io-app/assets/16268789/55e3b3a5-a0ff-46d4-8034-0b1c063a1451" />

This behavior in our actual codebase has been introduced by the version `3.0.7`.

File: package.json [hunk 2]
```diff
@@ -174,14 +175,14 @@
     "react-native-reanimated": "^2.9.1",
     "react-native-render-html": "^6.3.1",
     "react-native-responsive-screen": "^1.4.1",
-    "react-native-safe-area-context": "^3.3.2",
+    "react-native-safe-area-context": "^4.9.0",
     "react-native-screen-brightness": "^2.0.0-alpha",
-    "react-native-screens": "^2.18.1",
+    "react-native-screens": "^3.29.0",
```
Comment (shadowsheep1) [inline package.json:180]: @LazyAfternoons did you forgot to push the commit that revert this?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (LazyAfternoons) [reply package.json]: Sorry, the push failed and I didn't notice!

**Review by shadowsheep1 — APPROVED** [review]
LGTM, tested:
- FL
- Email uniqueness
- Profile Data Navigation
- Onboarding/First Onboarding navigation

**Pull request merged**